src/gui/transferlistmodel.cpp

```diff
@@ -658,7 +628,8 @@ QIcon getIconByState(const BitTorrent::TorrentState state)
     }
 }
 
-QColor getColorByState(const BitTorrent::TorrentState state)
+template <BitTorrent::TorrentState state>
```

Comment:
My eyes can't see it! 😱
Why it isn't regular function with regular parameter?

Reply:
I think we use a timer to update transfer list, so this code is called a lot even when the widget is not visible

Reply:
How it helps you?
If you want to really optimize it you need drop this function and move default color handing in getColorByState.

Reply:
>How it helps you? 

I suppose by using a template, the function will be easily inlined in `getColorByState`

> move default color handing in getColorByState.

there is a condition on which we assign colors and multiple torrent states share the same color, wouldn't moving it in `getColorByState` produce too redundancy and checks

Reply:
Seems you're wrong. You just create multiple similar functions (AFAIK, `switch` can't be executed in compile time even in modern C++). Or am I wrong?

Reply:
>switch can't be executed in compile time even in modern C++

Not sure, but I know `if constexpr ()` in C++17 is guaranteed to do it.

Anyway, I think this function shouldn't become a template, or at least not in current way.

Reply:
But there is no `switch constexpr`.
Anyway I strongly doubt template will give some tangible result in this case.